lib/language_pack/ruby_version.rb

```diff
@@ -87,18 +83,6 @@ def jruby?
       engine == :jruby
     end
 
-    # determine if we're using rbx
-    # @return [Boolean] true if we are and false if we aren't
-    def rbx?
-      engine == :rbx
-    end
-
-    # determines if a build ruby is required
-    # @return [Boolean] true if a build ruby is required
-    def build?
```

Comment:
Should this also be in the changelog? "Dropped support for building 1.8.7 and 1.9.2" or something? This seems like something that could be shipped independently from heroku-24/multi-arch?

Reply:
The oldest version of ruby on heroku-20 is 2.5 and heroku-22 is 3.1 (https://devcenter.heroku.com/articles/ruby-support#unsupported-ruby-versions). Ruby 1.9.2 was not available on heroku-18 either:

```
⛄️ 3.1.4 🚀 /Users/rschneeman/Documents/projects/rpm-Project-Code-gatech-kbai (main)
$ curl -I https://heroku-buildpack-ruby.s3.us-east-1.amazonaws.com/heroku-18/ruby-2.5.7.tgz
HTTP/1.1 200 OK
x-amz-id-2: Hl89PTav6LfvU90vP97YsBc5DEBptax9s4PZU+3ubIrIUBY54lD0f2FXMOCgh9XQEQRU+5ia1x4=
x-amz-request-id: KVDZM54WRBN5ASGT
Date: Mon, 15 Apr 2024 15:55:27 GMT
Last-Modified: Tue, 01 Oct 2019 14:46:30 GMT
ETag: "93ff97625abfc6ccd1a071bb0fda8e66"
x-amz-version-id: null
Accept-Ranges: bytes
Content-Type:
Server: AmazonS3
Content-Length: 11995820

⛄️ 3.1.4 🚀 /Users/rschneeman/Documents/projects/rpm-Project-Code-gatech-kbai (main)
$ curl -I https://heroku-buildpack-ruby.s3.us-east-1.amazonaws.com/heroku-18/ruby-1.9.2.tgz
HTTP/1.1 404 Not Found
x-amz-request-id: 1NKVW4VD3T94T71R
x-amz-id-2: tTow7CFW9BcWmAbkEWvrqPz5zJdsy7ry8Jm39v+EURa4RN/UpslBhhGf1iHzOrpJ4n1BJxiGGoI=
Content-Type: application/xml
Date: Mon, 15 Apr 2024 15:55:31 GMT
Server: AmazonS3
```

This is removing REALLY old cruft, so I don't think it needs a changelog. The rubinius stuff is in a similar situation. I have a habit of not touching even dead code branches in the buildpack, but keeping those unused paths around means that it's more resistant to change. I've come to understand from working with Ed how much of a liability not-cleaning up such paths can be. As this code was making my job harder, I decided that now was a good time to remove it.

In short: It's being removed right now because it was difficult to disentangle the prior download and build logic which touched these 2 other branches:

- Build rubies
- Rubinius

I'll try extracting it out to it's own PR, but this PR will be blocked on that one getting merged first.